frontend/src/components/stages/election_editor.ts

```diff
@@ -0,0 +1,130 @@
+import '../../pair-components/textarea';
+
+import {MobxLitElement} from '@adobe/lit-mobx';
+import {CSSResultGroup, html, nothing} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+
+import '@material/web/checkbox/checkbox.js';
+
+import {core} from '../../core/core';
+import {ExperimentEditor} from '../../services/experiment.editor';
+
+import {
+  ElectionStageConfig,
+  ElectionItem
+} from '@deliberation-lab/utils';
+
+import {styles} from './election_editor.scss';
+
+/** Editor for election stage. */
+@customElement('election-editor')
+export class ElectionEditorComponent extends MobxLitElement {
+  static override styles: CSSResultGroup = [styles];
+
+  private readonly experimentEditor = core.getService(ExperimentEditor);
+
+  @property() stage: ElectionStageConfig|undefined = undefined;
+
+  override render() {
+    if (this.stage === undefined) {
+      return nothing;
+    }
+
+    return html`
+      ${this.renderElectionSettings()}
+    `;
+  }
+
+  private renderElectionSettings() {
+    if (!this.stage) return;
+    const isParticipantElection = this.stage.isParticipantElection;
+
+    const updateElection = () => {
+      if (!this.stage) return;
+      const isParticipantElection = !this.stage.isParticipantElection;
+      this.experimentEditor.updateStage({ ...this.stage, isParticipantElection });
+    };
+
+    return html`
+      <div class="section">
+        <div class="title">Election Settings</div>
+        <div class="checkbox-wrapper">
+          <md-checkbox
+            touch-target="wrapper"
+            ?checked=${isParticipantElection}
+            ?disabled=${!this.experimentEditor.canEditStages}
+            @click=${updateElection}
+          >
+          </md-checkbox>
+          <div>
+            Election among participants
+          </div>
+        </div>
+        ${isParticipantElection ? nothing : this.renderElectionItems()}
+      </div>
+    `;
+  }
+
+  private renderElectionItems() {
+    if (!this.stage) return nothing;
+    const electionItems: ElectionItem[] = this.stage.electionItems || [];
+
+    const addItem = () => {
+      if (!this.stage) return;
+      const newId = String(electionItems.length + 1); // Generate a numeric ID as string
```

Comment:
Can we use `generateId` from `utils/shared` or some other random method? This is going to generate duplicates if an item is deleted

Reply:
could also add a `createElectionItem` function in `utils/stages/election_stage.ts`